src/cswinrt/strings/additions/Windows.Foundation/Windows.Foundation.cs

```diff
@@ -302,12 +302,12 @@ internal void CompleteFromAsyncActionWithProgress(IAsyncActionWithProgress<TProg
 
         internal void CompleteFromAsyncOperation(IAsyncOperation<TResult> asyncInfo, AsyncStatus asyncStatus)
         {
-            Complete(asyncInfo, ai => ((IAsyncOperation<TResult>)ai).GetResults(), asyncStatus);
+            Complete(asyncInfo, ai => ai.As<IAsyncOperation<TResult>>().GetResults(), asyncStatus);
```

Comment:
This `As<T>` overload is deprecated. You need to pass an IID.

Reply:
For IID, it would be like IAsyncOperationMethods<TResult>.IID.

Reply:
Here it's calling into https://github.com/microsoft/CsWinRT/blob/9854573b9880650efd87c5be41a553d125877c92/src/WinRT.Runtime/CastExtensions.cs#L29. I don't see where it is deprecated.

Reply:
Update: the change here is unnecessary. it's a leftover when I was debugging the issue.
Reverting the commit.